radixdlt-engine/src/main/java/com/radixdlt/atom/TxAction.java

```diff
@@ -64,5 +64,216 @@
 
 package com.radixdlt.atom;
 
-/** Marker interface for RadixEngine actions. */
-public interface TxAction {}
+import static com.radixdlt.utils.Strings.asEmptyIfNull;
+import static java.util.Objects.requireNonNull;
+
+import com.radixdlt.crypto.ECPublicKey;
+import com.radixdlt.identifiers.REAddr;
+import com.radixdlt.utils.Bytes;
+import com.radixdlt.utils.UInt256;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.function.LongFunction;
+
+/** RadixEngine actions. */
+public sealed interface TxAction {
```

Comment:
Quite a few of the actions under here are missing their custom ToString methods

Reply:
See comment above.